frontend/src/components/Order/OrderButton/OrderButton.tsx

```diff
@@ -3,36 +3,92 @@ import { useFormContext } from 'react-hook-form';
 import { useNavigate } from 'react-router-dom';
 import { renderOrderSuccessToast } from '@/utils/toastContents.tsx';
 import { toast } from 'react-toastify';
+import useFetchProductData from '@/hooks/fetch/useFetchProductData.ts';
+import { useEffect } from 'react';
+import { getUserInfo } from '../../../../storage/userInfo.ts';
+import order from '../../../../api/order.ts';
+import { PATH } from '@/constants/path';
 
 interface Props {
   id: number;
   count: number;
-  receiveForm: {
-    submittedRef: React.MutableRefObject<{ count: number }[] | null>;
+  receiverForm: {
+    submittedRef: React.MutableRefObject<{ name: string; phone: string; count: number }[] | null>;
   };
 }
 
-export default function OrderButton({ id, ranking, count, receiverForm }: Props) {
+export default function OrderButton({ id, count, receiverForm }: Props) {
   const { handleSubmit } = useFormContext();
   const { submittedRef } = receiverForm;
   const navigate = useNavigate();
+  const { product, loading, error } = useFetchProductData(id);
 
-  const item = ranking.find((item) => Number(item.id) === id);
-  const price = item.price.sellingPrice * count;
+  useEffect(() => {
+    if (error) {
+      toast.error(error, {
+        autoClose: 3000,
+      });
+      navigate(`${PATH.HOME}`);
+    }
+  }, [error]);
 
-  const onSubmit = (data) => {
-    if (!submittedRef.current || submittedRef.current.length === 0) return;
+  if (!product) {
+    return <div>상품 정보가 없습니다.</div>
+  }
 
-    const { textMessage, senderName } = data;
+  const price = product.price * count;
 
-    toast(renderOrderSuccessToast(item.name, count, senderName, textMessage), {
-      type: 'success',
-      autoClose: 3000,
-      style: { width: '400px' },
-    });
+  const onSubmit = async (data) => {
+    const userInfo = getUserInfo();
+    const token = userInfo?.authToken;
 
-    navigate('/');
+    if (!token) {
+      toast.error('로그인이 필요합니다.');
+      navigate(`${PATH.LOGIN}`);
+      return;
+    }
+
+    if (!submittedRef.current || submittedRef.current.length === 0) {
+      toast.error('받는 사람 정보를 입력해주세요.');
+      return;
+    }
+
+    const { textMessage, senderName, messageCardId } = data;
+
+    const orderData = {
+      productId: product.id,
+      message: textMessage,
+      messageCardId: String(messageCardId),
+      ordererName: senderName,
+      receivers: submittedRef.current.map(({ name, phone, count }) => ({
+        name,
+        phoneNumber: phone,
+        quantity: Number(count),
+      })),
+    }
+
+    try {
+      const result = await order(orderData);
+      if (result.data?.success) {
+        toast(renderOrderSuccessToast(product.name, count, senderName, textMessage), {
+          type: 'success',
+          autoClose: 3000,
+          style: { width: '400px' },
+        });
+        navigate(`${PATH.HOME}`);
+      }
+    } catch (error) {
+      toast.error(error.message || '주문 중 오류가 발생했습니다.');
```

Comment:
[타입] 조건문을 통해 타입을 개선해보면 좋아보여요

```suggestion
    } catch (error) {
      if (error instanceof Error) {
        toast.error(error.message);
      } else {
        toast.error('주문 중 오류가 발생했습니다.');
      }생했습니다.');
```

Reply:
error가 Error인지 아닌지에 따라 다르게 보이도록 수정하였습니다.